pkg/asset/installconfig/aws/validation.go

```diff
@@ -329,12 +329,8 @@ func validateEndpointAccessibility(endpointURL string) error {
 	if port == "" {
 		port = "https"
 	}
-	conn, err := proxy.Dial(context.Background(), "tcp", net.JoinHostPort(URL.Hostname(), port))
-	if err != nil {
-		return err
-	}
-	conn.Close()
-	return nil
+	_, err = http.Head(net.JoinHostPort(URL.Hostname(), port))
```

Comment:
I don't think that we need the part where we split `endpointURL` into hostname and port and then recombine them. In fact, we actually want to retain the other parts of the URL, especially the scheme.